hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java

```diff
@@ -1110,7 +1117,7 @@ public static RegionInfo getRegionInfo(final Result r, byte [] qualifier) {
   public static TableState getTableState(Connection conn, TableName tableName)
       throws IOException {
     if (tableName.equals(TableName.META_TABLE_NAME)) {
-      return new TableState(tableName, TableState.State.ENABLED);
+      throw new IllegalAccessError("Go to the Master to find hbase:meta table state, not here");
```

Comment:
So client trying to get the table state - will have to use some new API. Fine.

Reply:
Yes sir.

Master knows state for all tables and for meta so can ask the Master just for hbase:meta table state or you can ask Master for any table's state. Behind the scenes it proxies the request to this method if user-space tables. Otherwise, it gets state from memory (there is a note how hbase:meta table state is intentionally transient defaulting to ENABLED in case error).

Other implication noted is that this patch will add load to the Master. Master will now field requests for table state. It caches table state but these rpcs now go via the Master where before they went to host with meta table.

Thanks for taking look.